ibu-imager/ops/ops.go

```diff
@@ -67,3 +78,78 @@ func (o *ops) RunBashInHostNamespace(command string, args ...string) (string, er
 	args = append([]string{command}, args...)
 	return o.RunInHostNamespace("bash", "-c", strings.Join(args, " "))
 }
+
+// WaitForEtcd waits for the availability of the etcd server at the given healthz endpoint.
+// It continuously checks the healthz endpoint to determine the server's availability.
+// If the etcd server becomes accessible, it returns nil, indicating success. Otherwise, it returns an error.
+func (o *ops) WaitForEtcd(healthzEndpoint string) error {
+	// Use the default HTTP client
+	client := http.DefaultClient
+
+	for {
+		// Send an HTTP GET request to the healthz endpoint
+		resp, err := client.Get(healthzEndpoint)
+		if err != nil {
+			// If there was an error, wait for a short interval before retrying
+			time.Sleep(1 * time.Second)
+			o.log.Info(err)
+			continue
+		}
+
+		// Ensure the response body is closed after reading
+		defer resp.Body.Close()
+
+		// If the server responds with HTTP status 200 (OK), it's up and running
+		if resp.StatusCode == http.StatusOK {
+			return nil
+		}
+
+		// Wait for a short interval before retrying
+		time.Sleep(1 * time.Second)
+	}
+}
+
+// GetImageFromPodDefinition reads a YAML file containing pod configuration data
+// and extracts the image name for a given container named.
+// It returns the image name if found, or an error if not found or encountered any issues.
+func (o *ops) GetImageFromPodDefinition(containerImage string) (string, error) {
+
+	// Define a struct to match the structure of the YAML data
+	type PodConfig struct {
+		APIVersion string `yaml:"apiVersion"`
+		Kind       string `yaml:"kind"`
+		Metadata   struct {
+			Name string `yaml:"name"`
+		} `yaml:"metadata"`
+		Spec struct {
+			Containers []struct {
+				Name  string `yaml:"name"`
+				Image string `yaml:"image"`
+			} `yaml:"containers"`
+		} `yaml:"spec"`
+	}
+
+	// Read the YAML file
+	yamlData, err := os.ReadFile(etcdStaticPodFile)
```

Comment:
The function has a generic name, so it should receive `etcdStaticPodFile` as a parameter and not hardcode it.

This would make the call-site look like `GetImageFromPodDefinition(etcdStaticPodFile, "etcd")` which I feel is much easier to follow.

Also create a `var etcdStaticPodEtcdContainer string = "etcd"` global var so the call looks even clearer:

`GetImageFromPodDefinition(etcdStaticPodFile, etcdStaticPodEtcdContainer)`